lib/Dialect/MSFT/ExportQuartusTcl.cpp

```diff
@@ -37,18 +41,32 @@ struct TclOutputState {
     os.indent(2);
     return os;
   };
+
+  SmallVector<Attribute> symbolRefs;
 };
 } // anonymous namespace
 
-void emitPath(TclOutputState &s, RootedInstancePathAttr path) {
-  for (auto part : path.getPath())
-    s.os << part.getValue() << '|';
+void emitPath(TclOutputState &s, RootedInstancePathAttr path,
+              SymbolCache &symCache) {
+  for (auto part : path.getPath()) {
+    auto inst = cast<msft::InstanceOp>(symCache.getDefinition(part));
+    assert(inst && "path instance must be in symbol cache");
+    auto mod = inst->getParentOfType<MSFTModuleOp>();
+    assert(mod && "path instance must have MSFT module parent");
+
+    s.os << "{{" << s.symbolRefs.size() << "}}" << '|';
```

Comment:
Took me a minute to figure out what's going on in the rest of the lines. Maybe add a code comment?